### PR TITLE
Annual Calendar Extension v1. Base functionality for annual calendar generation

### PR DESCRIPTION
**Pull request by stringham2**
Create monthly calendar shapes for a given year & language.

v1 supports the following locales:
en-US, es-LA, de, ja, fr-FR, nl, pt-BR, ru, it-IT, pl-PL

### CODE REVIEW

File: annual-calendar-creator/public/calendarsettings.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        <link type="text/css" rel="stylesheet" href="./calendarsettings.css">
+    </head>
+    <body>
+
+        <script type="text/javascript">
+            function setCalendarSettings() {
+                let year = document.getElementById('calendar-year').value;
+                let language = document.getElementById('calendar-language').value;
+                
+                let localizedMonths = [1,2,3,4,5,6,7,8,9,10,11,12].map(m => new Intl.DateTimeFormat(language,{month: 'long', year: 'numeric'}).format(new Date(`${m}/01/${year}`)));
```
Comment (geoffmaddox) [inline annual-calendar-creator/public/calendarsettings.html]: I think ``new Date(`${m}/01/${year}`)`` would differ depending on the browser locale, so would `new Date(year, m - 1)` be safer to get the date here?

Edit: Same for the next line

Comment (stringham2) [reply annual-calendar-creator/public/calendarsettings.html]: Adjusted this, good catch!

**Review by geoffmaddox — APPROVED** [review]
Looks good to me! This is very neat.

File: annual-calendar-creator/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Annual Calendar Creator
+
+Annual Calendar Creator is a Lucid application that re-organizes shapes and elements on your documents into circles.
```
Comment (cball18) [inline annual-calendar-creator/README.md]: I'm assuming this is a description for a different extension?

Comment (jnowjack-lucidchart) [inline annual-calendar-creator/README.md]: > that re-organizes shapes and elements on your documents into circles.

I'm not 100% certain, but I don't think that's what this extension does :slightly_smiling_face: (similar phrasing exists in the `## Terms of use` section below too).

This file seems very similar to the `annual-calendar-creator/editorextensions/README.md` one. Are both needed? Can one just be a link to the other so you don't have to repeat yourself?

File: annual-calendar-creator/editorextensions/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Annual Calendar Generator Extension
+
+Users can create an annual calendar for a given year & language. This supports languages supported by Lucid as of June 2024, thoughmore languages could be added.
```
Comment (cball18) [inline annual-calendar-creator/editorextensions/README.md]: nit: though more

**Review by cball18 — APPROVED** [review]

File: annual-calendar-creator/README.md (hunk 2 above)
Comment (jselvaraaj) [reply annual-calendar-creator/README.md]: ☝️

File: annual-calendar-creator/editorextensions/README.md (hunk 3 above)
Comment (jselvaraaj) [reply annual-calendar-creator/editorextensions/README.md]: ☝️

File: annual-calendar-creator/editorextensions/annual-calendar-creator/src/extension.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            [TextMarkupNames.HAlign]: 'left',
+            [TextMarkupNames.Family]: TEXT_FONT_FAMILY
+        });
+    }
+}
+
+export function createAnnualCalendar(year: number, months: string[], days: string[]) {
+    let page = viewport.getCurrentPage();
+
+    let MONTHS: Month[] = months.map((name, index) => {
```
Comment (jselvaraaj) [inline annual-calendar-creator/editorextensions/annual-calendar-creator/src/extension.ts]: nit: prefer to have lowercase for variable names and uppercase for constants. could use `rawMonths` and `months`

**Review by jselvaraaj — APPROVED** [review]

File: annual-calendar-creator/editorextensions/annual-calendar-creator/src/calendarsettingsmodal.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { EditorClient, Modal } from 'lucid-extension-sdk';
+import { createAnnualCalendar } from './extension';
+
+export interface CalendarSettingsModalMessage {
+    'year': number;
+    'language': string;
+    'localizedMonths': string[],
+    'localizedDays': string[]
```
Comment (tjames-lucid) [inline annual-calendar-creator/editorextensions/annual-calendar-creator/src/calendarsettingsmodal.ts:8]: Some of the formatting here looks strange with inconsistencies between using ; and ,

Comment (stringham2) [reply annual-calendar-creator/editorextensions/annual-calendar-creator/src/calendarsettingsmodal.ts:8]: Good catch

File: annual-calendar-creator/README.md (hunk 2 above)
Comment (stringham2) [reply annual-calendar-creator/README.md]: Both are not needed, I'll remove one of them.

File: annual-calendar-creator/editorextensions/annual-calendar-creator/src/extension.ts (hunk 4 above)
Comment (stringham2) [reply annual-calendar-creator/editorextensions/annual-calendar-creator/src/extension.ts]: good point! Updated!

**Review by jselvaraaj — APPROVED** [review]

**Pull request merged**